TEMP

bryceosterhaus opened the pull request:
Hey, check out my commits on top of your work and let me know if this helps give you some direction.

kresimir-coko committed:
feat(@clayui/empty-state): create component

kresimir-coko committed:
feat(@clay-components): Input Move Boxes - Old

kresimir-coko commented on packages/clay-form/stories/index.tsx:
```diff
@@ -76,6 +76,16 @@ const moveBoxesOptions = [
+		// 	onChange: //
+		// 	selected: //
+		// 	onSelectChange: //
+		// }}
```
So this pattern is pretty, but I don't get what the value of `onChange` would be. Would the purpose, and the expectation, be that the dev who uses this component will be able to provide their own function?

bryceosterhaus replied:
@kresimir-coko  `onChange` would be the callback for whenever the items in the box are changed. It would be the same callback for if we just used the `SelectBox` component.

```jsx
<ClaySelectBox
	items: // items rendered in list
	onChange: // callback for when items are changed. Like when they are reordered(up/down).
	selected: // value for which items are selected
	onSelectChange: // callback for when the selected items change.
/>
```

Essentially we need a way to know when the items in the list are changed. They could change by being removed, added, or re-ordered.

Added/Removed: when an item is moved between boxes.
Reordered: when an item is moved up/down in the list.

kresimir-coko committed:
feat(@clay-components): Input Move Boxes - Add _select-box.scss and rename some selectors

kresimir-coko committed:
feat(@clay-components): Select Box

kresimir-coko committed:
feat(@clay-components): Input Move Boxes

kresimir-coko committed:
feat(@clay-components): Input Move Boxes & Select Box stories

kresimir-coko committed:
feat(@clay-components): Input Move Boxes - Make component fully controlled and add some more classes to be used in tests

kresimir-coko committed:
feat(@clay-components): Input Move Boxes - Tests and Snapshots, incomplete

bryceosterhaus committed:
feat(@clayui/modal): add low-level componenets for header

bryceosterhaus committed:
docs(clayui.com): add modal docs for Header

bryceosterhaus committed:
feat(clayui.com): use @clayui/css instead of clay-css

bryceosterhaus committed:
docs(clayui.com): wrap docs in ClayIconSpriteContext.Provider

pat270 committed:
fix(clay-css): Aspect Ratio `.aspect-ratio-item` should be aligned to the left when parent container has `text-align: center`

issue #833

pat270 committed:
feat(clay-css): Adds Empty State component

feat(clay-css): Empty State adds Sass maps `$c-empty-state`, `$c-empty-state-animation`, `$c-empty-state-image`, `$c-empty-state-aspect-ratio`, `$c-empty-state-title`, `$c-empty-state-text`, `$c-empty-state-footer`

issue #833

pat270 committed:
feat(clay-css): Adds images for Empty State in `src/images/images`

issue #833

pat270 committed:
chore(clay-css-test-site): Empty State adds examples

kresimir-coko committed:
feat(@clay-components): DualListbox - Rename and refactor from Input Move Boxes, update snapshots and markup

mohouyizme committed:
fix(CHANGELOG.md): fix a typo

pat270 committed:
feat(@clayui/css): Mixins `clay-label-variant` use `clay-css` mixin in each selector instead of individually declaring properties

feat(@clayui/css): Mixins `clay-label-variant` adds options to pass Sass maps `hover`, `focus`, `disabled`, `link`, `link-hover` which use `clay-css` mixin

fix(@clayui/css): Mixins `clay-label-variant` deprecate options `bg`, `border-color`, `color`, `outline`, `text-decoration`, `transition`, `hover-bg`, `hover-border-color`, `hover-color`, `hover-text-decoration`, `focus-bg`, `focus-border-color`, `focus-box-shadow`, `focus-color`, `focus-outline`, `focus-text-decoration`, `disabled-bg`, `disabled-border-color`, `disabled-box-shadow`, `disabled-color`, `link-color`, `link-text-decoration`, `link-hover-color`, `link-hover-text-decoration`

issue #2975

bryceosterhaus committed:
chore: add gh action for checking package size

bryceosterhaus committed:
Merge pull request #2982 from pat270/clay-833

feat(clay-css): Adds Empty State component

bryceosterhaus committed:
chore(@clayui/empty-state): run format

bryceosterhaus committed:
chore(@clayui/empty-state): move everything to index.tsx

pat270 committed:
feat(@clayui/css): Labels adds `label-inverse-*` to invert label colors

feat(@clayui/css): Labels adds Sass maps `$label-inverse-primary`, `$label-inverse-secondary`, `$label-inverse-success`, `$label-inverse-info`, `$label-inverse-warning`, `$label-inverse-danger`, `$label-inverse-light`, `$label-inverse-dark`

fixes #2975

bryceosterhaus committed:
chore(@clayui/empty-state): no need to have explicit container prop when we can just use ...otherProps

bryceosterhaus committed:
chore(clayui/empty-state): update snapshots

pat270 committed:
chore(clay-css-test-site): Labels add examples of `label-inverse-*`

bryceosterhaus committed:
update tests

bryceosterhaus committed:
rename from ClayReorder to ClaySelectBox

bryceosterhaus committed:
Merge pull request #2979 from bryceosterhaus/2844

feat(@clayui/modal): add low-level componenets for header

bryceosterhaus committed:
refactor API

bryceosterhaus committed:
allow controlled aspect of select to be optional

bryceosterhaus committed:
update tests

bryceosterhaus committed:
fix(@clayui/breadcrumb): use ClayLink over anchor tag

bryceosterhaus committed:
fix(@clayui/card): use ClayLink over anchor tag

bryceosterhaus committed:
fix(@clayui/drop-down): use ClayLink over anchor tag

bryceosterhaus committed:
fix(@clayui/label): use ClayLink over anchor tag

bryceosterhaus committed:
fix(@clayui/list): use ClayLink over anchor tag

bryceosterhaus committed:
fix(@clayui/pagination): use ClayLink over anchor tag

bryceosterhaus committed:
chore: update snapshots

bryceosterhaus committed:
Merge pull request #2987 from bryceosterhaus/master

chore: add gh action for checking package size

bryceosterhaus committed:
fix(@clayui/tooltip): add polyfill for element.matches

bryceosterhaus committed:
chore(@clayui/empty-state): for consistency with other components use imgSrc and add imgProps for accessibility

bryceosterhaus committed:
chore: use all caps for constants

bryceosterhaus committed:
Merge pull request #2989 from bryceosterhaus/pr-2937-copy

chore(clayui/empty-state): update snapshots

bryceosterhaus committed:
Merge pull request #3000 from bryceosterhaus/master

chore: use all caps for constants

bryceosterhaus committed:
chore: add verbose flag to debug github action

bryceosterhaus committed:
Merge pull request #3002 from bryceosterhaus/master

chore: add verbose flag to debug github action

bryceosterhaus committed:
chore: forgot backslash

bryceosterhaus committed:
Merge pull request #3003 from bryceosterhaus/master

chore: forgot backslash

bryceosterhaus committed:
chore: disable action due to it not working properly when PR is from a fork

bryceosterhaus committed:
Merge pull request #3004 from bryceosterhaus/master

chore: disable action due to it not working properly when PR is from a fork

bryceosterhaus committed:
Merge pull request #2986 from mohouyizme/master

fix(CHANGELOG.md): fix a typo

pat270 committed:
fix(@clayui/css): SVG Icons update the flag of Slovakia

fixes #2996

bryceosterhaus committed:
Merge pull request #2995 from bryceosterhaus/2991

fix: use ClayLink over anchor tag

bryceosterhaus committed:
Merge pull request #3006 from pat270/clay-2996

fix(@clayui/css): SVG Icons update the flag of Slovakia

pat270 committed:
fix(@clayui/css): Forms `input[type="range"]` focus border artifact in Chrome and remove padding

fixes #1179

bryceosterhaus committed:
Merge pull request #3007 from bryceosterhaus/2998

fix(@clayui/tooltip): add polyfill for element.matches

bryceosterhaus committed:
chore: update CONTRIBUTING.md

bryceosterhaus committed:
chore(release): publish updated packages

 - clayui.com@3.0.0-alpha.10
 - @clayui/autocomplete@3.1.1
 - @clayui/breadcrumb@3.1.1
 - @clayui/card@3.1.1
 - @clayui/color-picker@3.2.1
 - @clayui/css@3.8.0
 - @clayui/date-picker@3.1.1
 - @clayui/drop-down@3.4.0
 - @clayui/label@3.1.1
 - @clayui/list@3.2.1
 - @clayui/modal@3.4.0
 - @clayui/multi-select@3.5.1
 - @clayui/multi-step-nav@3.1.1
 - @clayui/pagination-bar@3.1.1
 - @clayui/pagination@3.1.1
 - @clayui/tooltip@3.2.1
 - demos@0.2.1

bryceosterhaus committed:
chore: fix gh action error

bryceosterhaus committed:
Merge pull request #3010 from bryceosterhaus/master

chore: fix gh action error

kresimir-coko commented:
Looked at it and included in the final, squashed, commit in the [main pr](https://github.com/liferay/clay/pull/2949)
